tidy3d/components/source.py

```diff
@@ -465,6 +465,29 @@ def _dir_vector(self) -> Tuple[float, float, float]:
         return None
 
 
+class BroadbandSource(Source, ABC):
+    """A source with frequency dependent field distributions."""
+
+    num_freqs: pydantic.NonNegativeInt = pydantic.Field(
+        1,
+        title="Number of Frequency Points",
+        description="Number of points to approximate the frequency dependence of injected field.",
+    )
+
+    @cached_property
+    def frequency_grid(self) -> np.ndarray:
+        """A Chebyshev grid used to approximate frequency dependence."""
+        fmin, fmax = self.source_time.frequency_range(4)
+        return 0.5 * (fmin + fmax) + 0.5 * (fmax - fmin) * np.cos(
```

Comment:
I think we could make this more readable by splitting it over a few lines and assigning some intermediate variables, for example:

```py
f_add = 0.5 * (fmin + fmax)
f_diff = 0.5 * (fmax - fmin)
f_range = (2 * np.flip(np.arange(self.num_freqs)) + 1) / self.num_freqs
return f_add + f_diff * np.cos(0.5 * np.pi * f_range)
```

or something like this (double check..)